cmake/EthOptions.cmake

```diff
@@ -1,7 +1,9 @@
 macro(configure_project)
 	# Default to RelWithDebInfo configuration if no configuration is explicitly specified.
-	set(CMAKE_BUILD_TYPE RelWithDebInfo CACHE STRING "Build type on single-configuration generators")
-
+	if(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
```

Comment:
You are somehow right. We should not set CMAKE_CONFIGURATION_TYPES at all, because it affects only Visual Studio.
You can find it in the main CMakeLists.txt.